Wikipedia/Code/QuadKey.swift

```diff
@@ -0,0 +1,298 @@
+//https://msdn.microsoft.com/en-us/library/bb259689.aspx
+//http://wiki.openstreetmap.org/wiki/QuadTiles
+
+public typealias QuadKey = UInt64
+public typealias QuadKeyPart = UInt32
+public typealias QuadKeyPrecision = UInt16
+public typealias QuadKeyDegrees = Double
+
+public extension QuadKeyPrecision {
+    public static let maxPrecision: QuadKeyPrecision = 32
```

Comment:
so there are two bits necessary to store each precision level, 32 total fit in the 64 bit integer of the quad key - this is the maximum precision we can store

the view controller uses 16 as the max because that's the max level that we care about grouping locations together - below that we'd rather all individual pins be visible